### ISSUE TITLE
Using `use_shared_state::<Collection>(cx)` withut subscribing to current context

### ISSUE THREAD

**Issue by serzhiio** [DioxusLabs/dioxus#864]
Each time i use `use_shared_state::<Collection>(cx).unwrap();` it subscribes current context for update notification on each `.write()` call.
But i do not want such behavior everywhere. Let's say i have a  coroutine whose purpose is only to update data and notify all subscribers about it. 
I think this is a good feature.

**Comment by Demonthos** [comment]
One thing that Fermi does is split read and writes so that a component that only writes a shared value does not need to re-render when another component writes to the value. There are many other more advanced features like selectors that Fermi can support, but I'm not sure if they belong in use_shared_state. use_shared_state is a minimal shared state hook. It is a similar starting point for state management, but it is very limited. For more advanced features like this Fermi should be used instead

**Comment by serzhiio** [comment]
Ok, let's say i have `let workers_mpub = use_atom_ref(cx, WORKERS_MPUB);`.
and when I do `workers_mpub.write().update(...data...);` later on atomic ref it calls:
```impl<T: 'static> UseAtomRef<T> {
    pub fn write(&self) -> RefMut<T> {
        self.root.force_update(self.ptr);  <---- this calls rerender
        self.value.borrow_mut()
    }
```
and `use_atom_ref()` itself subscribe current context to all changes.
It means everywhere I use `use_atomic_ref()` to get link to Atom it will rerender all contexts subscribed. Am I wrong?

**Comment by Demonthos** [comment]
> 

Most times when you use a global state Fermi will subscribe to the component so that writes to the state trigger rerenders for the component. The exception I was talking about is the [use_set hook](https://docs.rs/fermi/latest/fermi/hooks/fn.use_set.html). When you call the hook it will not subscribe to the state because we know that it will only ever be written to, not read

**Comment by serzhiio** [comment]
Resolved after #1035
